internal/pkg/mapper/sharedcode/links/rename.go

```diff
@@ -0,0 +1,91 @@
+package links
+
+import (
+	"context"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils"
+)
+
+func (m *mapper) OnObjectsRename(event model.OnObjectsRenameEvent) error {
+	errors := utils.NewMultiError()
+
+	// Find renamed shared codes
+	renamedSharedCodes := make(map[string]bool)
+	for _, object := range event.RenamedObjects {
+		// Is config?
+		if record, ok := object.Record.(*model.ConfigManifest); ok {
+			// Get component
+			component, err := m.State.Components().Get(record.ComponentKey())
+			if err != nil {
+				errors.Append(err)
+				continue
+			}
+
+			// Is shared code?
+			if component.IsSharedCode() {
+				renamedSharedCodes[record.Key().String()] = true
+			}
+		}
+	}
+
+	// Find configs using these shared codes
+	uow := m.NewUnitOfWork(context.Background())
+	for _, objectState := range m.State.All() {
+		configState, err := m.getDependentConfig(objectState, renamedSharedCodes)
+		if err != nil {
+			errors.Append(err)
+		} else if configState == nil {
+			continue
+		}
+
+		// Re-save config -> new "shared_code_path" will be saved.
+		m.Logger.Debugf(`Updating "shared_code_path" in "%s"`, configState.Path())
+		uow.SaveObject(configState, configState.Local, model.NewChangedFields("configuration"))
+	}
+
```

Comment:
Tym, ze sa pouzila `map`, to mame spravene pomocou 2x `for` a nemuseli sa pouzit ziadne vnorene cykly -> je to tak rychlejsie.